ddtrace/contrib/tornado/futures.py

```diff
@@ -0,0 +1,38 @@
+from ddtrace import tracer
+from ddtrace.context import Context
+
+
+def _wrap_submit(func, instance, args, kwargs):
+    """
+    Wrap `Executor` method used to submit a work executed in another
+    thread. This wrapper ensures that a new `Context` is created and
+    properly propagated using an intermediate function.
+    """
+    # create a new Context with the right active Span
+    # TODO: the current implementation doesn't provide the GlobalTracer
+    # singleton, so we should rely in our top-level import
+    ctx = Context()
+    current_ctx = tracer.context_provider.active()
+    if current_ctx is not None:
+        ctx._current_span = current_ctx.get_current_span()
```

Comment:
can't use a shallow copy (`current_ctx.copy()`) because we must not copy the `trace` array or the number of opened spans, otherwise we have a race condition on the array + will never flush the trace.

We can isolate that behavior in a specific method of the `Context` though, so we can use that generic method all around the code. Will do that later in another PR.

Reply:
Right, I was more thinking about a custom `copy` method (forgot that Python had a default one). So we should probably find another name. And that one would only copy `_current_span` / traceID / spanID / priority / sampled.

Reply:
Oh ok, gotcha. Yes that's the plan.